packages/e2e/src/load-test-scheduler.ts

```diff
@@ -0,0 +1,30 @@
+import { Logger } from 'ts-log';
+import { Observable, map, mergeMap, take, timer } from 'rxjs';
+
+export interface LoadTestSchedulerProps<T> {
+  duration: number;
+  callsPerDuration: number;
+  idOffset?: number;
+  callUnderTest: (id: number) => Observable<T>;
+}
+
+export interface LoadTestSchedulerDependencies {
+  logger: Logger;
+}
+
+export const getLoadTestScheduler = <T>(
+  { duration, callsPerDuration, idOffset = 0, callUnderTest }: LoadTestSchedulerProps<T>,
+  { logger }: LoadTestSchedulerDependencies
+): Observable<T> =>
+  new Observable<T>((observer) => {
+    const callIntervalMs = (duration * 1000) / callsPerDuration;
+    logger.info(`Running ${callsPerDuration} calls in ${duration}s, 1 call / ${callIntervalMs}ms`);
+
+    return timer(0, callIntervalMs)
+      .pipe(
+        map((id) => id + idOffset),
+        take(callsPerDuration),
+        mergeMap((id) => callUnderTest(id))
+      )
+      .subscribe(observer);
```

Comment:
Do we know how this behaves in comparison with artillery? Is it use a linear progression of calls under test or any other approach to spread the load for the given timeframe/duration? (sorry about my lack of knowledge around rxjs)

Reply:
This is a simple linear progression.

Reply:
I'm not sure if it's a linear progression though, because it has `take(callsPerDuration)` which means it sends a static number of requests (VUs) per duration (they do not increase over time), which way is the same as artillery's `arrivalCount`.

To achieve a linear load progression, we must mimic the `rampTo` [rate](https://www.artillery.io/docs/guides/guides/test-script-reference#ramp-up-rate) which is the linear ramp-up phase where the number of new arrivals increases linearly over time. @mirceahasegan

Reply:
I think you're right. I find the `rampTo` rate a bit confusing regarding to the total sessions started during the test.
In this case, the requirement is to init a number of wallets in a period of time (evenly split imho).
I would leave it like this if there isn't a specific request to have a ramping number of values started each interval.

Reply:
Yep, this current load-test-scheduler implementation is fine with the wallet init load test requirements (as we are satisfied with a static total count of wallets to be initialized).

Might be it's a note that in other test cases we definitely would need to extend the scheduler to support different load-spreading options based on the configuration and env variables, same as artillery )